subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy

```diff
@@ -62,49 +62,52 @@ import static org.hamcrest.CoreMatchers.containsString
 
 class BuildServiceIntegrationTest extends AbstractIntegrationSpec {
 
-    def "does not nag when service is used by task without a corresponding usesService call and feature preview is NOT enabled"() {
+    def "nagging for undeclared service usage"() {
```

Comment:
This data-driven test replaces the multiple variants we had; with the additional condition on whether a service actually specified a `maxParallelUsages` constraint, having dedicated test cases was getting too messy.